SVDBasisGenerator.h

```diff
@@ -108,7 +108,8 @@ class SVDBasisGenerator
              d_svdsampler->isNewTimeInterval()) {
             d_svdsampler->resetDt(dt);
             if (d_basis_writer) {
-                d_basis_writer->writeBasis();
+                d_basis_writer->writeBasis("basis");
+                writeSnapshot();
```

Comment:
I've added this line to write the snapshot matrix as well whenever a time interval ends. It doesn't do the last one though, so the user will still have to call writeSnapshot for the last one, similar to calling endSamples. Do we want this?